src/worker/plugins/run.ts

```diff
@@ -19,8 +25,16 @@ export async function runPlugins(server: PluginsServer, event: PluginEvent): Pro
                 await processError(server, pluginConfig, error, returnedEvent)
                 server.statsd?.increment(`plugin.${pluginConfig.plugin?.name}.process_event.ERROR`)
             }
-            server.statsd?.timing(`plugin.${pluginConfig.plugin?.name}.process_event`, timer)
 
+            const timeSpentRunning = new Date().getTime() - timer.getTime()
+            server.statsd?.timing(`plugin.${pluginConfig.plugin?.name}.process_event`, timer)
+            posthog.capture('$plugin_running_duration', {
+                team: event.team_id,
+                plugin: pluginConfig.plugin?.name,
+                time_ms: timeSpentRunning,
+                time_seconds: Math.round(timeSpentRunning / 10) / 1000,
+                function: 'processEvent',
+            })
```

Comment:
This effectively doubles the amount of events we emit. For capturing aggregated metrics, that's way too much volume.

Reply:
I'd rather aggregate this information and return it in an instance status report, perhaps once per hour or whenever. We could also add the instance ID into this report to help deduping.